src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -1222,12 +1239,16 @@ def delete_service_principal_credential(cmd, identifier, key_id, cert=False):
 
 
 def _resolve_service_principal(client, identifier):
-    # todo: confirm with graph team that a service principal name must be unique
```

Comment:
This question remains unanswered. If a service principal name is not unique, some extra complexity may occur.

Reply:
I can confirm: a `servicePrincipalName` value uniquely identifies a **servicePrincipal** object within a tenant. In other words `$filter=servicePrincipalNames/any(n:n eq '{something')` will always return exactly 0 or 1 values.